modules/EvseManager/IECStateMachine.cpp

```diff
@@ -0,0 +1,451 @@
+        return CPEvent::MREC_24_ConnectorVoltageHigh;
+    case types::board_support_common::Event::MREC_25_BrokenLatch:
+        return CPEvent::MREC_25_BrokenLatch;
+    case types::board_support_common::Event::MREC_26_CutCable:
+        return CPEvent::MREC_26_CutCable;
+    default:
+        return RawCPState::Disabled;
+    }
+}
+
+/// \brief Converts the given Event \p e to human readable string
+/// \returns a string representation of the Event
+const std::string cpevent_to_string(CPEvent e) {
+    switch (e) {
+    case CPEvent::CarPluggedIn:
+        return "CarPluggedIn";
+    case CPEvent::CarRequestedPower:
+        return "CarRequestedPower";
+    case CPEvent::PowerOn:
+        return "PowerOn";
+    case CPEvent::PowerOff:
+        return "PowerOff";
+    case CPEvent::CarRequestedStopPower:
+        return "CarRequestedStopPower";
+    case CPEvent::CarUnplugged:
+        return "CarUnplugged";
+    case CPEvent::ErrorE:
+        return "ErrorE";
+    case CPEvent::ErrorDF:
+        return "ErrorDF";
+    case CPEvent::ErrorVentilationNotAvailable:
+        return "ErrorVentilationNotAvailable";
+    case CPEvent::ErrorBrownOut:
+        return "ErrorBrownOut";
+    case CPEvent::ErrorEnergyManagement:
+        return "ErrorEnergyManagement";
+    case CPEvent::EFtoBCD:
+        return "EFtoBCD";
+    case CPEvent::BCDtoEF:
+        return "BCDtoEF";
+    case CPEvent::PermanentFault:
+        return "PermanentFault";
+    case CPEvent::EvseReplugStarted:
+        return "EvseReplugStarted";
+    case CPEvent::EvseReplugFinished:
+        return "EvseReplugFinished";
+    case CPEvent::MREC_2_GroundFailure:
+        return "MREC_2_GroundFailure";
+    case CPEvent::MREC_3_HighTemperature:
+        return "MREC_3_HighTemperature";
+    case CPEvent::MREC_4_OverCurrentFailure:
+        return "MREC_4_OverCurrentFailure";
+    case CPEvent::MREC_5_OverVoltage:
+        return "MREC_5_OverVoltage";
+    case CPEvent::MREC_6_UnderVoltage:
+        return "MREC_6_UnderVoltage";
+    case CPEvent::MREC_8_EmergencyStop:
+        return "MREC_8_EmergencyStop";
+    case CPEvent::MREC_15_PowerLoss:
+        return "MREC_15_PowerLoss";
+    case CPEvent::MREC_17_EVSEContactorFault:
+        return "MREC_17_EVSEContactorFault";
+    case CPEvent::MREC_18_CableOverTempDerate:
+        return "MREC_18_CableOverTempDerate";
+    case CPEvent::MREC_19_CableOverTempStop:
+        return "MREC_19_CableOverTempStop";
+    case CPEvent::MREC_20_PartialInsertion:
+        return "MREC_20_PartialInsertion";
+    case CPEvent::MREC_23_ProximityFault:
+        return "MREC_23_ProximityFault";
+    case CPEvent::MREC_24_ConnectorVoltageHigh:
+        return "MREC_24_ConnectorVoltageHigh";
+    case CPEvent::MREC_25_BrokenLatch:
+        return "MREC_25_BrokenLatch";
+    case CPEvent::MREC_26_CutCable:
+        return "MREC_26_CutCable";
+    }
+    throw std::out_of_range("No known string conversion for provided enum of type CPEvent");
```

Comment:
The compiler should be able to check if we have covered all options for an enum class

Reply:
Some compilers will give a warning if ont enum value is missing. Still something needs to happen if it is the case anyway